express-notes-api/index.js

```diff
@@ -0,0 +1,87 @@
+const data = require('./data');
+const fs = require('fs');
+
+var express = require('express');
+var app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const dataArray = [];
+  for (const key in data.notes) {
+    dataArray.push(data.notes[key]);
+  }
+  res.json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  // const numID = Number(req.params.id);
+  if (!Number.isInteger(req.params.id) && req.params.id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!data.notes[req.params.id]) {
+    res.status(404).json({ error: 'cannot find note with this id' });
+  } else {
+    res.status(200);
+    res.send(data.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes/', (req, res) => {
+  if (!req.body.content) {
+    res.status(400).json({ error: 'content is a required field' });
+  } else if (req.body.content) {
+    const noteObj = req.body;
+    noteObj.id = data.nextId;
+    data.notes[noteObj.id] = noteObj;
+    data.nextId++;
+    fs.writeFile('data.json', JSON.stringify(data, null, 2), err => {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      }
+    });
+    res.status(201);
+    res.send(noteObj);
```

Comment:
`writeFile` is asynchronous: please make sure your response is sent _after_ `writeFile` finishes without error.